File: greedy/python/cookies.py
class Solution:
    def findContentChildren(self, g, s):
        g.sort()
        s.sort()

        j, ans = 0, 0
        # Para cada criança
        for i in range(len(g)):
            # Tenta encontrar um cookie que se encaixe
            while j < len(s) and s[j] < g[i]:
                j += 1
            # Se não houver mais cookies
            if j == len(s):
                break
            else:
                ans += 1
                j += 1

        return ans
    
s = Solution()
print(s.findContentChildren([1, 2, 3], [1, 1]))
print(s.findContentChildren([1, 2], [2, 3, 1]))

File: greedy/cpp/cookies.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());

        int j = 0, ans = 0;
        // for each child
        for(int i = 0; i < g.size(); i++) {
            // try to find a cookie that fits
            while(j < s.size() && s[j] < g[i]){
                j++;
            }
            // if there's no more cookies
            if(j == s.size())
                break;
            else {
                ans++;
                j++;
            }
        }

        return ans;
    }
};

int main() {
    // read children
    int n_children;
    cin >> n_children;
    vector<int> g(n_children);
    for(int i = 0; i < n_children; i++) {
        cin >> g[i];
    }

    // read cookies
    int n_cookies;
    cin >> n_cookies;
    vector<int> s(n_cookies);
    for(int i = 0; i < n_cookies; i++) {
        cin >> s[i];
    }

    Solution solve;
    cout << solve.findContentChildren(g, s) << endl;
}